pkg/autoscaler/aggregation/bucketing.go

```diff
@@ -60,22 +60,32 @@ func (t *TimedFloat64Buckets) IsEmpty() bool {
 	return len(t.buckets) == 0
 }
 
-// GetAndLock returns the buckets and secures them via a mutex. The contents of
-// the returned map can be modified and the modifications will be visible to other
-// readers. Unlock must be called after the client is done reading/manipulating
-// the data.
-func (t *TimedFloat64Buckets) GetAndLock() map[time.Time]Float64Bucket {
-	t.bucketsMutex.Lock()
-	return t.buckets
+// ForEachBucket calls the given Accumulator function for each bucket.
+func (t *TimedFloat64Buckets) ForEachBucket(accs ...Accumulator) {
```

Comment:
I'd go with just `ForEach` since we're operating on buckets, but that's fine as is.